src/cargo/core/manifest.rs

```diff
@@ -16,14 +16,23 @@ use crate::core::{Dependency, PackageId, PackageIdSpec, SourceId, Summary};
 use crate::core::{Edition, Feature, Features, WorkspaceConfig};
 use crate::util::errors::*;
 use crate::util::interning::InternedString;
-use crate::util::toml::{TomlManifest, TomlProfiles};
+use crate::util::toml::{DefinedTomlManifest, TomlProfiles};
 use crate::util::{short_hash, Config, Filesystem};
 
 pub enum EitherManifest {
     Real(Manifest),
     Virtual(VirtualManifest),
 }
 
+impl EitherManifest {
+    pub fn workspace(&self) -> Rc<WorkspaceConfig> {
```

Comment:
FWIW I see a number of methods here returning a bare `Rc<T>`, but if possible I think it would be best to return `&Rc<T>` since it defers the clone to the point at which it's actually necessary.

Reply:
This isn't really a perf concern since the reference counting is almost nothing in Cargo, mostly just a stylistic thing.